core/pta/bcm/hw_rng.c

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2019, Broadcom
+ */
+
+#include <drivers/bcm_hwrng.h>
+#include <io.h>
+#include <kernel/pseudo_ta.h>
+#include <trace.h>
+
+#define HW_RNG_SERVICE_UUID \
+		{ 0x6272636D, 0x2019, 0x0201,  \
+		{ 0x42, 0x43, 0x4D, 0x5F, 0x52, 0x4E, 0x47, 0x30 } }
+
+#define PTA_BCM_HW_RNG_CMD_GET	0
+
+#define HW_RNG_TA_NAME		"pta_hwrng.ta"
+
+static TEE_Result pta_hw_rng_get(uint32_t param_types,
+				 TEE_Param params[TEE_NUM_PARAMS])
+{
+	uint32_t num_words = 0;
+	uint32_t *rnd_num = NULL;
+	uint32_t res = 0;
+	uint32_t exp_param_types = TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_OUTPUT,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE,
+						   TEE_PARAM_TYPE_NONE);
+
+	if (exp_param_types != param_types) {
+		EMSG("Invalid Param types");
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	rnd_num = (uint32_t *)&params[0].value.a;
```

Comment:
64-bit support? Why not use `memref.buffer`? Is `value.a` supposed to be a physical address? Missing `tee_mmu_check_access_rights()`.

Reply:
`value.a` just holds the 32-bit HW generated random number. Will update to use local variable

Reply:
Ah, yes, my bad. I dunno why I thought it was a memory reference :-/ Yeah, a simple assignment would look better I think.